Sources/Starknet/Signers/StarkCurveSigner.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+public class StarkCurveSigner: StarknetSignerProtocol {
+    private let privateKey: Felt
+
+    public private(set) lazy var publicKey: Felt? = getPublicKey()
```

Comment:
I think for consumers this is not ideal, in theory you can initialise a signer with an invalid private key and not catching it until the moment you try access the public key, instead maybe we could have a failable initialiser where it throws or returns nil if we are not able to get the public key.

Reply:
Good point

Reply:
Updated